src/Builder/JsonExpressionBuilder.php

```diff
@@ -65,8 +63,10 @@ public function build(ExpressionInterface $expression, array &$params = []): str
 
     /**
      * @return string The typecast expression based on {@see JsonExpression::getType()}.
+     *
+     * @psalm-param JsonExpression $expression
      */
-    protected function getTypecast(JsonExpression $expression): string
+    private function getTypecast(ExpressionInterface $expression): string
```

Comment:
Keep `JsonExpression`. If check `$expression` in `build()` then here will be `JsonExpression` guaranteed.